File: src/main/java/com/ddd/trainings/proposal/domain/ReviewerId.java
package com.ddd.trainings.proposal.domain;

import com.ddd.utils.BaseId;
import java.util.UUID;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@EqualsAndHashCode
@ToString
public class ReviewerId extends BaseId {

  @Getter
  private final String value;

  public ReviewerId(String value) {
    this.value = value;
  }

  public static ReviewerId create() {
    return new ReviewerId(UUID.randomUUID().toString());
  }
}

File: src/Ddd/Trainings/Proposal/Domain/ReviewerId.cs
using System;
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Domain;

public class ReviewerId(string value) : BaseId
{
    private readonly string _value = value;

    public static ReviewerId Create()
    {
        return new ReviewerId(Guid.NewGuid().ToString());
    }

  public override bool Equals(object? obj)
  {
    return obj is ReviewerId id &&
           _value == id._value;
  }

  public override int GetHashCode()
  {
    return HashCode.Combine(_value);
  }

  public override string GetValue()
    {
        return _value;
    }
} 